### PR TITLE
Update Cache In-Memory to 6.0

### LINKED ISSUE #23725: .NET 6: Cache in-memory
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 13f5f8f9-18dd-9c12-9ed1-fcc6b44ba7fd
* Version Independent ID: 640b4f36-5886-7b69-1c95-cfc8574a02fd
* Content: [Cache in-memory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-6.0)
* Content Source: [aspnetcore/performance/caching/memory.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/performance/caching/memory.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #23725.

[Internal Review](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-6.0&branch=pr-en-us-23866).

### CODE REVIEW

File: aspnetcore/performance/caching/memory.md [hunk 1]
```diff
@@ -150,39 +143,40 @@ The size of the cache entry can be set by <xref:Microsoft.Extensions.Caching.Mem
 * Items with the earliest absolute expiration.
 * Items with the earliest sliding expiration.
 
-Pinned items with priority <xref:Microsoft.Extensions.Caching.Memory.CacheItemPriority.NeverRemove> are never removed. The following code removes a cache item and calls `Compact`:
+Pinned items with priority <xref:Microsoft.Extensions.Caching.Memory.CacheItemPriority.NeverRemove> are never removed. The following code removes a cache item and calls `Compact` to remove 25% of cached entries:
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/memory.md:146]: Is my never link a distraction or useful? You have a similar link below. Never used to be a suggestion and in some caches it's a suggestion. When memory gets too low, never is trimmed in some caches, but not this one.

```suggestion
Pinned items with priority <xref:Microsoft.Extensions.Caching.Memory.CacheItemPriority.NeverRemove> are [never removed](https://github.com/dotnet/runtime/blob/release/6.0/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs#L395-L427). The following code removes a cache item and calls `Compact` to remove 25% of cached entries:
```

**Review by Rick-Anderson — APPROVED** [review]
Wow, really great rewrite. Much easier to follow.

File: aspnetcore/performance/caching/memory.md (hunk 1 above)
Comment (serpent5) [reply aspnetcore/performance/caching/memory.md:146]: Yes, I think it's helpful. I'll commit the suggestion, but just focus the range slightly to cover the switch that decides what the priorities mean. Thanks!

**Comment by Rick-Anderson** [comment]
@serpent5  I'm going to S&M then merge to live. If you need additional changes you'll have to do another PR. This content is too good to leave in a PR.

**Pull request merged**